go/teams/service_helper.go

```diff
@@ -659,20 +659,38 @@ func AcceptInvite(ctx context.Context, g *libkb.GlobalContext, token string) err
 	return err
 }
 
-func ParseAndAcceptSeitanToken(ctx context.Context, g *libkb.GlobalContext, tok string) (wasSeitan bool, err error) {
+func parseAndAcceptSeitanTokenV1(ctx context.Context, g *libkb.GlobalContext, tok string) (wasSeitan bool, err error) {
 	seitan, err := GenerateIKeyFromString(tok)
 	if err != nil {
 		g.Log.CDebugf(ctx, "GenerateIKeyFromString error: %s", err)
 		g.Log.CDebugf(ctx, "returning TeamInviteBadToken instead")
 		return false, libkb.TeamInviteBadTokenError{}
 	}
-	g.Log.CDebugf(ctx, "found seitan token")
 	err = AcceptSeitan(ctx, g, seitan)
+	return true, err
+}
+
+func parseAndAcceptSeitanTokenV2(ctx context.Context, g *libkb.GlobalContext, tok string) (wasSeitan bool, err error) {
+	seitan, err := GenerateIKeyV2FromString(tok)
 	if err != nil {
-		return true, err
+		g.Log.CDebugf(ctx, "GenerateIKeyV2FromString error: %s, tok: %v", err, tok)
+		g.Log.CDebugf(ctx, "returning TeamInviteBadToken instead")
+		return false, libkb.TeamInviteBadTokenError{}
 	}
+	err = AcceptSeitanV2(ctx, g, seitan)
+	return true, err
+
+}
 
-	return true, nil
+func ParseAndAcceptSeitanToken(ctx context.Context, g *libkb.GlobalContext, tok string) (wasSeitan bool, err error) {
+	wasSeitan, err = parseAndAcceptSeitanTokenV1(ctx, g, tok)
+	if err != nil {
+		wasSeitan, err = parseAndAcceptSeitanTokenV2(ctx, g, tok)
+	}
+	if wasSeitan {
+		g.Log.CDebugf(ctx, "found seitan token")
+	}
+	return true, err
```

Comment:
this is a bit confusing for me, should it be `return wasSeitan, err` ?

but good that it tries to handle both versions

Reply:
yes! this was a bug. fixed.